$OpenBSD: patch-dip_c,v 1.2 2012/02/27 18:21:41 fgsch Exp $

From John Bowler <jbowler@acm.org> via NetBSD pkg/44940:

There is *NO* error handling, no call to setjmp();
so when png_error is called the call stack ends up destroyed and,
apparently, the program dies in create_read_struct_2, right after
the comment that explains why libpng is about to call abort() ;-)

--- dip.c.orig	Sat Jul  7 23:23:39 2007
+++ dip.c	Thu Jul 21 20:31:14 2011
@@ -1436,6 +1436,8 @@ unsigned char *png_data, int png_length, struct style 
 	
 	png_ptr=png_create_read_struct(PNG_LIBPNG_VER_STRING,
 			NULL, my_png_error, my_png_warning);
+	if (setjmp(png_jmpbuf(png_ptr)))
+	    overalloc(); /* some error detected by libpng */
 	info_ptr=png_create_info_struct(png_ptr);
 	png_set_read_fn(png_ptr,&work,(png_rw_ptr)&read_stored_data);
 	png_read_info(png_ptr, info_ptr);
